zio-http/shared/src/main/scala/zio/http/internal/FormState.scala

```diff
@@ -29,8 +29,11 @@ private[http] object FormState {
 
   final class FormStateBuffer(boundary: Boundary) extends FormState { self =>
 
-    private val tree0: ChunkBuilder[FormAST] = ChunkBuilder.make[FormAST]()
-    private val buffer: ChunkBuilder.Byte    = new ChunkBuilder.Byte
+    private val tree0: ChunkBuilder[FormAST]    = ChunkBuilder.make[FormAST]()
+    private val buffer: ChunkBuilder[Byte]      = new ChunkBuilder.Byte
+    private var bufferSize: Int                 = 0
+    private val closingBoundaryBytesSize        = boundary.closingBoundaryBytes.size
+    private var boundaryMatches: Array[Boolean] = new Array[Boolean](closingBoundaryBytesSize)
```

Comment:
There is something here that I don't understand / failing to see. What is the reason for using an array? Can't this be just a boolean and then L62-68 become this?

```scala
boundaryMatches &&= posInLine < closingBoundaryBytesSize && boundary.closingBoundaryBytes(posInLine) == byte

val boundaryDetected = boundaryMatches && posInLine == closingBoundaryBytesSize - 1
```

Reply:
Wonderful, you are right we do not need to remember all checked bytes this way.

That work => https://github.com/zio/zio-http/pull/2862/commits/f1f33ba98abc12714d706d0f3a89496c62b082c1

Thanks.

Reply:
tbh no idea why i thought we need an array :see_no_evil: so much better without - nice find @kyri-petrou :tada: